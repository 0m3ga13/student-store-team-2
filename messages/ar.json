{
    "Index": {
        "Title": "المتجر الصغير",
        "Home": "الرئيسية",
        "Products": "المنتجات",
        "Orders": "الطلبيات",
        "State": "الحالة",
        "Buy": "شراء",
        "Borrow": "إعارة",
        "Description": "هذه المنصة هي مربحة للجميع: القرضاء يحققون دخلاً إضافيًا، والمقترضون يقللون التكاليف.",
        "Profile": "الحسـاب",
        "For Sale": "للبيع",
        "For Borrow": "للإعارة",
        "Sale & Borrow": "إعارة وشراء",
        "Day": "لليوم",
        "How It Works": "طريـقة العمل",
        "Hello": "مرحبا",
        "Sign out": "تسجيل الخروج",
        "Sign In": "تسجيل الدخول",
        "Login with Google": "سحل الدخول بحسابك",
        "Sign Up With Google": "سحل الدخول بحسابك"
    }
}
