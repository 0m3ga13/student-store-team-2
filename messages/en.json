{
    "Index": {
        "Title": "MiniStore",
        "Home": "Home",
        "Products": "Products",
        "Orders": "Orders",
        "State": "State",
        "Donate": "Donate",
        "Buy": "Buy",
        "Borrow": "Borrow",
        "Description": "This platform is a win-win for everyone involved lenders earn extra income, borrowers cut costs.",
        "Profile": "Profile",
        "for_sale": "for_sale",
        "for_borrow": "for_borrow",
        "Sale & Borrow": "Sale & Borrow",
        "Day": "Day",
        "How It Works": "How It Works",
        "recently added": "recently added",
        "Hello": "Hello",
        "Sign out": "Sign out",
        "Sign In": "Sign In",
        "Login with Google": "Login with Google",
        "Sign Up With Google": "Sign Up With Google",
        "Free delivery": "Free delivery",
        "feature descriprion 1": "You can receive your order without incurring any additional shipping costs",
        "qualite guarantee": "qualite guarantee",
        "feature descriprion 2": "Our products or services meet high standards and perform as promised.",
        "daily offers": "daily offers",
        "feature descriprion 3": "Exclusive discounts and promotions on a daily basis",
        "100% secure payement": "100% secure payement",
        "feature descriprion 4": "Your financial information is protected and your transactions are safe from unauthorized access",
        "My Dashboard": "My Dashboard",
        "Awaiting Pickup": "Awaiting Pickup",
        "All": "All",
        "Active": "Active",
        "Completed": "Completed",
        "Sort": "Sort:",
        "Newest": "Newest",
        "Oldest": "Oldest",
        "my stuff":"my stuff",
        "borrowings": "borrowings",
        "lendings": "lendings",
        "purchases": "purchases",
        "sales": "sales",
        "My Borrowings": "My Borrowings",
        "My Lendings": "My Lendings",
        "My Purchases": "My Purchases",
        "My Sales": "My Sales",
        "General details": "General details",
        "First Name": "First Name",
        "Last Name": "Last Name",
        "Birth Date": "Birth Date",
        "Gender": "Gender",
        "Institution": "Institution",
        "Contact & address Detail": "Contact & address Detail",
        "Phone Number": "Phone Number",
        "Building": "Building",
        "Street": "Street",
        "City": "City",
        "State_place": "State",
        "Save Changes": "Save Changes",
        "Upload Image": "Upload Image",
        "Close": "Close",
        "Upload": "Upload",
        "Loading": "Loading",
        "Not a member ?": "Not a member ?",
        "Sign Up": "Sign Up",
        "Store": "Store",
        "Create Your": "Create Your",
        "and start exchanging with other students": "and start exchanging with other students",
        "Or Login Using:": "Or Login Using:",
        "Login": "Login",
        "LOGIN": "LOGIN ",
        "Already a member ?": "Already a member ?",
        "Email and password are required": "Email and password are required",
        "Password should be at least 8 characters long": "Password should be at least 8 characters long",
        "and Conditions and Privacy Policy of The website": "and Conditions and Privacy Policy of The website",
        "Email": "Email",
        "Password": "Password",
        "Full Name": "Full Name",
        "Address": "Address",
        "and Privacy Policy": "and Privacy Policy",
        "I have read and agree with all of the Terms": "I have read and agree with all of the Terms",
        "PROFILE": "PROFILE",
        "SECURITY": "SECURITY",
        "Upload ID": "Upload ID",
        "Edit ID": "Edit ID",
        "Edit Image": "Edit Image",
        "WHY DONATE": "WHY DONATE",
        "MAKE A DIFFERENCE BY DONATING": "MAKE A DIFFERENCE BY DONATING",
        "Donate paragraph": "In our fast-paced lives, it's easy to forget the power of gratitude. Students, often on tight budgets, rely on our support. Your donation is more than just financial aid; it's a symbol of gratitude. By giving generously, you're helping students create a brighter future, one contribution at a time.",
        "Small Help": "Small Help",
        "Some Help": "Some Help",
        "Big Help": "Big Help",
        "List an Item": "List an Item",
        "Drag and drop some files here or click to select files": "Drag and drop some files here or click to select files",
        "Location": "Location",
        "Category": "Category",
        "books": "books",
        "kitchen": "kitchen",
        "sports": "sports",
        "games": "games",
        "clothes": "clothes",
        "electronics": "electronics",
        "furniture": "furniture",
        "description": "Description",
        "Product Condition": "Product Condition",
        "Sell": "Sell",
        "Price Per Week": "Price Per Week",
        "Price": "Price",
        "Upload Image(s)": "Upload Image(s)",
        "List": "List",
        "Product Name": "Product Name",
        "Comments": "Comments",
        "Type your comment": "Type your comment",
        "Type your reply": "Type your reply",
        "Respond": "Respond",
        "Like": "Like",
        "Send": "Send",
        "Authentication error": "Invalid login credentials. Please check your email and password.",
        "Filter": "Filter",
        "Pending": "Pending",
        "Quick Links": "Quick Links",
        "About LACITÉ": "ABOUT LACITÉ",
        "About Us": "ABOUT US",
        "Github Project Link": "GITHUB PROJECT LINK",
        "Contact Us": "Contact Us",
        "We Couldn't Find the page you're looking for": "We Couldn't Find the page you're looking for",
        "Go To Home": "Go To Home",
        "We Couldn't Find the requested Product": "We Couldn't Find the requested Product."
    }
}
