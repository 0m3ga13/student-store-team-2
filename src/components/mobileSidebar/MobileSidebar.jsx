import Image from "next/image";
import Link from "next/link";
import { useTranslations } from "next-intl";
import { BiCollapseAlt } from "react-icons/bi";
import { CiLogin, CiLogout } from "react-icons/ci";
import { FaFacebook, FaGithub, FaLinkedin } from "react-icons/fa";
import { IoSettingsSharp } from "react-icons/io5";
import { MdAdminPanelSettings } from "react-icons/md";
import { PiInstagramLogo } from "react-icons/pi";

import { signOut } from "@/lib/_supabaseAuth";
import useTextDirection from "@/hooks/useTextDirection";

import LanguageChanger from "../languageChanger/LanguageChanger";
import { useUser } from "../userProvider/UserProvider";

export default function MobileSidebar({ toggleMobileMenu, isOpen }) {
    const t = useTranslations("Index");
    const direction = useTextDirection();

    const { user, userData, loading } = useUser();

    const handleSignOutUser = async () => {
        await signOut();

        window.location.href = "/";
    };

    return (
        <div className=' transition  animate-wiggle font-lato w-screen h-screen'>
            <div className='relative flex h-16 items-center justify-between text-titleContent'>
                {/* Welcome Message */}
                <div className='flex items-center ml-4'>
                    {user && (
                        <>
                            <Image
                                src={userData.avatar_url}
                                alt='Profile'
                                className='w-8 h-8 rounded-full'
                                width={50}
                                height={50}
                            />
                            <p className='ml-2 text-xl font-medium'>
                                {t("Welcome")} ,{userData.full_name}
                            </p>
                            <hr className=' border-gray-300 mx-4' />
                        </>
                    )}
                </div>
                {/* Toggle Button */}
                <div
                    className={`absolute inset-y-0 flex items-center sm:hidden ${
                        direction === "ltr" ? "right-0" : "left-0"
                    }`}
                >
                    <button
                        onClick={toggleMobileMenu}
                        className={`absolute ${
                            direction === "ltr" ? "right-0" : "left-0"
                        } items-center justify-center rounded-md p-2 hover:bg-accent hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white`}
                    >
                        <BiCollapseAlt
                            className='block h-8 w-8'
                            aria-hidden='true'
                        />
                    </button>
                </div>
            </div>

            {/* Navigation Links */}
            {user ? (
                <div className='space-y-1 mt-5 px-2 pt-2 pb-3 flex flex-col justify-start'>
                    <Link href='/profile?page=form'>
                        <div
                            className='hover:bg-accent hover:text-white rounded-md px-3 py-2 text-xl font-medium tracking-widest'
                            onClick={toggleMobileMenu}
                        >
                            <IoSettingsSharp className='inline-block mr-4 h-6 w-6 text-accent' />
                            {t("Profile")}
                        </div>
                    </Link>
                    <Link href='/profile?page=security'>
                        <div
                            className='hover:bg-accent hover:text-white rounded-md px-3 py-2 text-xl font-medium tracking-widest'
                            onClick={toggleMobileMenu}
                        >
                            <MdAdminPanelSettings className='inline-block mr-4 h-6 w-6 text-accent' />
                            {t("Security")}
                        </div>
                    </Link>

                    {/* Logout Button */}
                    <div
                        onClick={handleSignOutUser}
                        className='rounded-md px-3 py-2 text-xl font-medium tracking-widest flex items-center transform transition-colors duration-200 border-r-4 border-transparent hover:border-red-600 text-red-600'
                    >
                        <CiLogout className='inline-block mr-4 h-6 w-6' />

                        {t("Logout")}
                    </div>
                </div>
            ) : (
                <div className='space-y-1 mt-5 px-2 pt-2 pb-3 flex flex-col justify-start'>
                    <Link
                        href='/sign-in'
                        onClick={toggleMobileMenu}
                        className='hover:bg-accent hover:text-white rounded-md
                        px-3 py-2 text-xl font-medium tracking-widest'
                    >
                        <CiLogin className='inline-block mr-4 h-6 w-6' />
                        {t("Log In")}
                    </Link>
                </div>
            )}

            <hr className='mt-3 border-gray-300 mx-4' />

            {/*Other Links */}
            <div className='space-y-1 px-2 pt-2 pb-3 flex flex-col justify-start '>
                <div className='w-full lg:w-4/12 ml-auto px-3 py-2 text-md font-medium tracking-widest '>
                    <span className='block uppercase   font-semibold mb-2'>
                        {t("Quick Links")}
                    </span>
                    <ul className='list-unstyled'>
                        <li>
                            <Link
                                className='text-titleContent hover:text-accent font-semibold block pb-2 '
                                href='/'
                            >
                                {t("Home")}
                            </Link>
                        </li>
                        <li>
                            <Link
                                className='text-titleContent hover:text-accent font-semibold block pb-2 '
                                href='/products?page=1'
                            >
                                {t("Products")}
                            </Link>
                        </li>
                        <li>
                            <Link
                                className='text-titleContent hover:text-accent font-semibold block pb-2 '
                                href='/donate'
                            >
                                {t("Donate")}
                            </Link>
                        </li>
                    </ul>
                    <span className='block uppercase   font-semibold mb-2'>
                        {t("About LACITÃ‰")}
                    </span>
                    <ul className='list-unstyled'>
                        <li>
                            <Link
                                onClick={toggleMobileMenu}
                                href='https://github.com/202306-NEA-DZ-FEW/student-store-team-2'
                                target='_blank'
                                rel='noopener noreferrer'
                                className='text-titleContent hover:text-accent font-semibold block pb-2 '
                            >
                                {t("Github Project Link")}
                            </Link>
                        </li>
                        <li>
                            <Link
                                onClick={toggleMobileMenu}
                                className='text-titleContent hover:text-accent font-semibold block pb-2 '
                                href='/#how-it-works'
                            >
                                {t("How It Works")}
                            </Link>
                        </li>
                        <li>
                            <Link
                                onClick={toggleMobileMenu}
                                className='text-titleContent hover:text-accent font-semibold block pb-2 '
                                href='/about'
                            >
                                {t("About Us")}
                            </Link>
                        </li>
                    </ul>
                </div>
            </div>

            {/* Footer Links */}
            <div className='mt-auto px-2 pb-3'>
                <div
                    className={`flex flex-col ${
                        direction === "rtl"
                            ? "text-right lg:text-right"
                            : "text-left lg:text-left"
                    }`}
                >
                    <LanguageChanger />
                </div>
            </div>
            <div className='w-full flex  justify-center items-end'>
                <Link
                    type='button'
                    href='https://www.instagram.com/recodedofficial/'
                    target='_blank'
                    rel='noopener noreferrer'
                >
                    <PiInstagramLogo className='bg-white text-pink-400 shadow-lg font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none mr-2  ' />
                </Link>
                <Link
                    type='button'
                    href='https://www.facebook.com/recodedofficial/'
                    target='_blank'
                    rel='noopener noreferrer'
                >
                    <FaFacebook className='bg-white text-blue-900 shadow-lg font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none mr-2 ' />
                </Link>
                <Link
                    type='button'
                    href='https://www.linkedin.com/school/re-coded/'
                    target='_blank'
                    rel='noopener noreferrer'
                >
                    <FaLinkedin className='bg-white text-blue-400 shadow-lg font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none mr-2 ' />{" "}
                </Link>
                <Link
                    type='button'
                    href='https://github.com/202306-NEA-DZ-FEW/'
                    target='_blank'
                    rel='noopener noreferrer'
                >
                    <FaGithub className='bg-white text-black shadow-lg font-normal h-10 w-10 items-center justify-center align-center rounded-full outline-none focus:outline-none mr-2 ' />
                </Link>
            </div>
        </div>
    );
}
