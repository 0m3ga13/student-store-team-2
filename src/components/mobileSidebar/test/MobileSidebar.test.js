import renderer from "react-test-renderer";
import MobileSidebar from "../MobileSidebar";
import { NextIntlClientProvider } from "next-intl";
import messages from "../../../../messages/en.json";

jest.mock("next/navigation");
const pushMock = jest.fn();

it("renders correctly", () => {
    // Create a mock function for the 'navigation' prop
    const navigationMock = [];

    const tree = renderer
        .create(
            <NextIntlClientProvider locale='en' messages={messages}>
                <MobileSidebar navigation={navigationMock} />
            </NextIntlClientProvider>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
