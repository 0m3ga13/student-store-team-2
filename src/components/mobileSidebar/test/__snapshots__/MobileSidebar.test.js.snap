// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="transition  z-50  animate-wiggle
            "
  onKeyDown={[Function]}
>
  <div
    className=" relative flex h-16 items-center justify-between text-titleContent "
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="flex flex-shrink-0 items-center"
      >
        <h1
          className="tracking-widest font-lato font-semibold text-2xl px-5"
        >
          MiniStore
        </h1>
      </div>
    </a>
    <div
      className="absolute inset-y-0 right-0 flex items-center sm:hidden "
    >
      <button
        className="absolute right-0 items-center justify-center rounded-md p-2 hover:bg-accent hover:text-white focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
      >
        <svg
          aria-hidden="true"
          className="block h-8 w-8"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M2 15h7v7h2v-9H2v2zM15 2h-2v9h9V9h-7V2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="space-y-1 px-2 pt-2 pb-3 flex flex-col justify-start "
  >
    <a
      className="  hover:bg-accent hover:text-white rounded-md px-3 py-2 text-xl font-medium "
      href="/profile"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <button
        className="w-full flex justify-start items-center "
      >
        <svg
          aria-hidden="true"
          className="block h-6 w-6 mr-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 2a5 5 0 1 0 5 5 5 5 0 0 0-5-5zm0 8a3 3 0 1 1 3-3 3 3 0 0 1-3 3zm9 11v-1a7 7 0 0 0-7-7h-4a7 7 0 0 0-7 7v1h2v-1a5 5 0 0 1 5-5h4a5 5 0 0 1 5 5v1z"
          />
        </svg>
        <h2>
          Profile
        </h2>
      </button>
    </a>
    <div
      className=" items-center"
    >
      <div
        className="mx-auto max-w-md cursor-pointer "
        onKeyDown={[Function]}
      >
        <form
          action=""
          className="fixed left-5 sm:relative  mx-auto"
        >
          <input
            className=" cursor-pointer absolute  h-8 w-0  rounded-full  bg-transparent outline-none transition-all duration-300 ease-in-out left-10 pl-1"
            onChange={[Function]}
            type="search"
            value=""
          />
          <svg
            aria-hidden="true"
            className="mr-2 z-10 cursor-pointer my-auto h-6 w-6 stroke-gray-500  hover:text-accent rounded-xl font-semibold false"
            fill="currentColor"
            height="1em"
            onClick={[Function]}
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 18a7.952 7.952 0 0 0 4.897-1.688l4.396 4.396 1.414-1.414-4.396-4.396A7.952 7.952 0 0 0 18 10c0-4.411-3.589-8-8-8s-8 3.589-8 8 3.589 8 8 8zm0-14c3.309 0 6 2.691 6 6s-2.691 6-6 6-6-2.691-6-6 2.691-6 6-6z"
            />
            <path
              d="M11.412 8.586c.379.38.588.882.588 1.414h2a3.977 3.977 0 0 0-1.174-2.828c-1.514-1.512-4.139-1.512-5.652 0l1.412 1.416c.76-.758 2.07-.756 2.826-.002z"
            />
          </svg>
        </form>
      </div>
    </div>
  </div>
</div>
`;
